pkg/api/vminfo.go

```diff
@@ -337,29 +337,92 @@ func (b ResourceBounds) validate(ec *erc.Collector, path string, memSlotSize *re
 // ScalingConfig provides bits of configuration for how the autoscaler-agent makes scaling decisions
 type ScalingConfig struct {
 	// LoadAverageFractionTarget sets the desired fraction of current CPU that the load average
-	// should be. For example, with a value of 0.7, we'd want load average to sit at 0.7 ×
-	// CPU,
+	// should be. For example, with a value of 0.7, we'd want load average to sit at 0.7 × CPU,
 	// scaling CPU to make this happen.
-	LoadAverageFractionTarget float64 `json:"loadAverageFractionTarget"`
+	//
+	// When specifying the autoscaler-agent config, this field is required. For an individual VM, if
+	// this field is left out the settings will fall back on the global default.
+	LoadAverageFractionTarget *float64 `json:"loadAverageFractionTarget,omitempty"`
 
 	// MemoryUsageFractionTarget sets the desired fraction of current memory that
 	// we would like to be using. For example, with a value of 0.7, on a 4GB VM
 	// we'd like to be using 2.8GB of memory.
-	MemoryUsageFractionTarget float64 `json:"memoryUsageFractionTarget"`
+	//
+	// When specifying the autoscaler-agent config, this field is required. For an individual VM, if
+	// this field is left out the settings will fall back on the global default.
+	MemoryUsageFractionTarget *float64 `json:"memoryUsageFractionTarget,omitempty"`
+
+	// EnableLFCMetrics, if true, enables fetching additional metrics about the Local File Cache
+	// (LFC) to provide as input to the scaling algorithm.
+	//
+	// When specifying the autoscaler-agent config, this field is required. False is a safe default.
+	// For an individual VM, if this field is left out the settings will fall back on the global
+	// default.
+	EnableLFCMetrics *bool `json:"enableLFCMetrics,omitempty"`
 }
 
-func (c *ScalingConfig) Validate() error {
+// WithOverrides returns a new copy of defaults, where fields set in overrides replace the ones in
+// defaults but all others remain the same.
+//
+// overrides may be nil; if so, this method just returns defaults.
+func (defaults ScalingConfig) WithOverrides(overrides *ScalingConfig) ScalingConfig {
+	if overrides == nil {
+		return defaults
+	}
+
+	if overrides.LoadAverageFractionTarget != nil {
+		defaults.LoadAverageFractionTarget = &[]float64{*overrides.LoadAverageFractionTarget}[0] // slice trick to shallow copy
+	}
+	if overrides.MemoryUsageFractionTarget != nil {
+		defaults.MemoryUsageFractionTarget = &[]float64{*overrides.MemoryUsageFractionTarget}[0]
+	}
+	if overrides.EnableLFCMetrics != nil {
+		defaults.EnableLFCMetrics = &[]bool{*overrides.EnableLFCMetrics}[0]
```

Comment:
Shall we use `lo.ToPtr`?

Reply:
Ah yup, meant to do this - thanks for the reminder :) Implemented in #948 f9ed5d27408b77231d1e77d6338c6f248f939970